lib/core/src/lib/card-view/components/card-view-textitem/card-view-textitem.component.spec.ts

```diff
@@ -476,123 +474,112 @@ describe('CardViewTextItemComponent', () => {
+            updateTextField(component.property.key, 'updated-value');
+            await setTimeoutPromise(600);
+
+            fixture.detectChanges();
+            await fixture.whenStable();
+
+            expect(cardViewUpdateService.update).toHaveBeenCalledWith(property, 'updated-value');
+        });
 
-        it('should NOT trigger the update event if the editedValue is invalid', fakeAsync((done) => {
+        it('should NOT trigger the update event if the editedValue is invalid', async () => {
             component.property.isValid = () => false;
+            fixture.detectChanges();
+            await fixture.whenStable();
 
             fixture.detectChanges();
-            fixture.whenStable().then(() => {
-                fixture.detectChanges();
-                const cardViewUpdateService = TestBed.inject(CardViewUpdateService);
+            const cardViewUpdateService = TestBed.inject(CardViewUpdateService);
 
-                spyOn(cardViewUpdateService, 'update');
-                updateTextField(component.property.key, 'updated-value');
-                tick(600);
-                fixture.detectChanges();
-                fixture.whenStable().then(() => {
-                    expect(cardViewUpdateService.update).not.toHaveBeenCalled();
-                    done();
-                });
-            });
-        }));
+            spyOn(cardViewUpdateService, 'update');
+            updateTextField(component.property.key, 'updated-value');
 
-        it('should set the errorMessages properly if the editedValue is invalid', fakeAsync((done) => {
+            fixture.detectChanges();
+
+            await fixture.whenStable();
+            expect(cardViewUpdateService.update).not.toHaveBeenCalled();
+        });
+
+        it('should set the errorMessages properly if the editedValue is invalid', async () => {
             const expectedErrorMessages = [{ message: 'Something went wrong' } as CardViewItemValidator];
             component.property.isValid = () => false;
             component.property.getValidationErrors = () => expectedErrorMessages;
 
             fixture.detectChanges();
-            fixture.whenStable().then(() => {
+            await fixture.whenStable();
 
-                updateTextField(component.property.key, 'updated-value');
-                tick(600);
-                fixture.detectChanges();
-                fixture.whenStable().then(() => {
-                    expect(component.errors).toBe(expectedErrorMessages);
-                    done();
-                });
-            });
-        }));
+            updateTextField(component.property.key, 'updated-value');
+
+            fixture.detectChanges();
+            await fixture.whenStable();
+            expect(component.errors).toBe(expectedErrorMessages);
+        });
 
-        it('should render the error when the editedValue is invalid', fakeAsync((done) => {
+        it('should render the error when the editedValue is invalid', async () => {
```

Comment:
I've tested a few options on this test and only my solution works correctly. 
![image](https://user-images.githubusercontent.com/64163335/191693779-c9559813-c0ea-4d4d-abc9-8b24daa226e7.png)
![image](https://user-images.githubusercontent.com/64163335/191693841-5b1e95d9-f601-42df-bf4d-8128a5d392f4.png)
![image](https://user-images.githubusercontent.com/64163335/191693873-b1a55fc2-24c1-4663-9e01-747637849361.png)
![image](https://user-images.githubusercontent.com/64163335/191693967-485a38db-0c22-45d6-ab7f-b950a606f1b1.png)
![image](https://user-images.githubusercontent.com/64163335/191694019-e1894bcc-f9ca-405e-8f3b-ed2fb565b7a9.png)